WD-Firmware/comrv/comrv.c

```diff
@@ -18,34 +18,767 @@
+      uiProfilingIndication = D_COMRV_NO_LOAD_AND_RETURN_IND;
+#endif /* D_COMRV_FW_INSTRUMENTATION */
+   }
+
+#ifdef D_COMRV_MULTI_GROUP_SUPPORT
+   /* if the requested token isn't a multi-group token */
+   if (unToken.stFields.multiGroup == 0)
+   {
+#endif /* D_COMRV_MULTI_GROUP_SUPPORT */
+      /* search for token */
+      pAddress = comrvSearchForLoadedOverlayGroup(unToken);
+#ifdef D_COMRV_MULTI_GROUP_SUPPORT
+   }
+   /* search for a multi-group overlay token */
+   else
+   {
+      /* first ucEntryIndex to search from in the multi group table is determined by the overlayGroupID
+         field of the requested token */
+      ucEntryIndex = unToken.stFields.overlayGroupID;
+      do
+      {
+         /* search for the token */
+         pAddress = comrvSearchForLoadedOverlayGroup(pOverlayMultiGroupTokensTable[ucEntryIndex++]);
+      /* continue the search as long as the group wasn't found and we have additional tokens */
+      } while ((pAddress == NULL) && (pOverlayMultiGroupTokensTable[ucEntryIndex].uiValue != 0));
+
+      /* save the selected multi group entry */
+      usSelectedMultiGroupEntry = ucEntryIndex-1;
+   }
+#endif /* D_COMRV_MULTI_GROUP_SUPPORT */
+
+   /* get the group size */
+   usOverlayGroupSize = M_COMRV_GET_OVL_GROUP_SIZE(unToken);
+
+   /* if the data/function is not loaded we need to evict and load it */
+   if (pAddress == NULL)
+   {
+#ifdef D_COMRV_MULTI_GROUP_SUPPORT
+      /* if the requested token is a multi-group token */
+      if (unToken.stFields.multiGroup != 0)
+      {
+         /* for now we take the first token in the list of tokens */
+         unToken = pOverlayMultiGroupTokensTable[unToken.stFields.overlayGroupID];
+         /* save the selected multi group entry */
+         usSelectedMultiGroupEntry = unToken.stFields.overlayGroupID;
+      }
+#endif /* D_COMRV_MULTI_GROUP_SUPPORT */
+
+      /* disable ints */
+      // TODO: disable ints
+
+      /* get eviction candidates according to the requested pOverlayGroupSize */
+      ucNumOfEvictionCandidates = comrvGetEvictionCandidates(usOverlayGroupSize, ucEvictCandidateList);
+
+      ucEntryIndex = 0;
+      /* we need to handle heap fragmentation since we got more
+         than 1 eviction candidate */
+      if (ucNumOfEvictionCandidates > 1)
+      {
+         /* no handling to the last eviction candidate */
+         ucNumOfEvictionCandidates--;
+         while (ucEntryIndex < ucNumOfEvictionCandidates)
+         {
+            /* get the candidate entry */
+            pEntry = &stComrvCB.stOverlayCache[ucEvictCandidateList[ucEntryIndex]];
+            /* calc the source address */
+            pAddress = ((u08_t*)pEntry->pFixedEntryAddress + M_COMRV_GET_OVL_GROUP_SIZE_IN_BYTES(pEntry->unToken));
+            /* perform code copy */
+            comrvMemcpyHook(pEntry->pFixedEntryAddress, pAddress,
+                  stComrvCB.stOverlayCache[ucEvictCandidateList[ucEntryIndex]].pFixedEntryAddress - pAddress);
```

Comment:
stComrvCB.stOverlayCache[ucEvictCandidateList[ucEntryIndex]].pFixedEntryAddress 
assign local "meaningful" arg.

Reply:
done in 90f5569